src/App.js

```diff
@@ -8,24 +8,40 @@ import Decks from './pages/Decks';
 
 import { Routes, Route } from 'react-router-dom';
 import { useState, useEffect } from 'react';
+import useSWR from 'swr';
 import { nanoid } from 'nanoid';
 import { saveToLocal, loadFromLocal } from './utils/localStorage';
-import useCards from './hooks/useCards';
-
+import usePersonalCards from './hooks/useCards';
 import { ToastContainer } from 'react-toastify';
 
+const fetcher = (...args) => fetch(...args).then(res => res.json());
+
 function App() {
   const [allCategories, setAllCategories] = useState(
     loadFromLocal('allCategories') ?? []
   );
   const [showModal, setShowModal] = useState(false);
   const [currentId, setCurrentId] = useState('');
-  const { cards, setCards } = useCards();
+  const [personalCardsIds, setPersonalCardsIds] = useState(
+    loadFromLocal('personalCardsIds') ?? []
+  );
+  const { personalCards, setPersonalCards } = usePersonalCards();
```

Comment:
What is `personalCardIds` and `personalCards`?

Reply:
Explanation for all your questions: 
I had a tricky challenge: I want more than one user to use one backend together, like a study group. They share the information on the cards, so that they are able to split up the work.
BUT: If one of them pinns a card it shant be pinnt for the others.. Same with the right-wrong-counters..

I solved this in the following way:
The cards on the database are `publicCards`. They contain only question, answer and categories. They are fetched on every device and there is added 'isPinned', the counters and everything, that should be unique for every user in the state `personalCards`. I also save the ID of the card in this process in `personalCardIds` so that it's possible to add the cards only once.

Because of that it's also possible to delete cards only from the device with `handleDeleteCard` or to delete them from the database with `handleDeleteFromDatabase`.

It's all a bit complicated, but it works the way I want 🙂 

The only thing I want to improve is: At the moment the cards are fetched with a 'delay' of one card, because of the updating of the state.

I want to fix this with a 'synchronize'-button  🙂